### PR TITLE
Query parameter to list endpoints newest-to-oldest

### PR DESCRIPTION
**Pull request by svix-andor**
The overall pagination is still done first-to-last, but within a page results are listed last-to-first.

Builds on top of #812

### CODE REVIEW

File: server/svix-server/src/v1/endpoints/attempt.rs [hunk 1]
```diff
@@ -18,10 +18,10 @@ use crate::{
     v1::{
```
Comment (tasn) [inline server/svix-server/src/v1/endpoints/attempt.rs:18]: A lot of unrelated stuff

File: server/svix-server/src/v1/endpoints/endpoint/crud.rs [hunk 2]
```diff
@@ -58,9 +60,11 @@ pub(super) async fn list_endpoints(
     })
     .collect();
 
-    Ok(Json(EndpointOut::list_response_no_prev(
+    Ok(Json(EndpointOut::list_response_asc(
         results,
         limit as usize,
+        is_prev,
+        newest_first.unwrap_or(false),
```
Comment (tasn) [inline server/svix-server/src/v1/endpoints/endpoint/crud.rs:67]: Why not set a default in the newest_first structure directly and have it non-null?

File: server/svix-server/src/v1/endpoints/endpoint/crud.rs [hunk 3]
```diff
@@ -21,29 +21,31 @@ use crate::{
...
     },
     AppState,
 };
 use hack::EventTypeNameResult;
 
 pub(super) async fn list_endpoints(
     State(AppState { ref db, .. }): State<AppState>,
-    pagination: ValidatedQuery<Pagination<EndpointId>>,
+    ValidatedQuery(pagination): ValidatedQuery<Pagination<ReversibleIterator<EndpointId>>>,
+    ValidatedQuery(NewestFirst { newest_first }): ValidatedQuery<NewestFirst>,
```
Comment (tasn) [inline server/svix-server/src/v1/endpoints/endpoint/crud.rs:36]: Also, I think we can probably come up with a better name. It was just a terrible place-holder I suggested. :P

File: server/svix-server/src/v1/endpoints/attempt.rs (hunk 1 above)
Comment (svix-andor) [reply server/svix-server/src/v1/endpoints/attempt.rs]: unrelated stuff gone after rebase :)

File: server/svix-server/src/v1/endpoints/endpoint/crud.rs (hunk 2 above)
Comment (svix-andor) [reply server/svix-server/src/v1/endpoints/endpoint/crud.rs:67]: This way the struct remains more generic and can be reused elsewhere where default `true` is needed, and number of lines-wise it ends up being more or less the same. But I don't feel super strong about this, if you think having a default on the struct itself is significantly better I'm ok with changing to that.

File: server/svix-server/src/v1/utils/mod.rs [hunk 4]
```diff
@@ -288,7 +311,14 @@ pub trait ModelOut: Clone {
...
             IteratorDirection::Next
         };
-        list_response_inner(data, limit, ListOrdering::Descending, direction, true)
+        list_response_inner(
+            data,
+            limit,
+            ListOrdering::Descending,
+            direction,
+            true,
+            false,
```
Comment (tasn) [inline server/svix-server/src/v1/utils/mod.rs:320]: 1. This is going out of control with this true, false, and a million params.
2. Luckily it's not necessarily. `newest_first` == order descending. You don't need another param, you already have it. If it's newest first, you just do descending order...

File: server/svix-server/src/v1/endpoints/endpoint/crud.rs (hunk 3 above)
Comment (svix-andor) [reply server/svix-server/src/v1/endpoints/endpoint/crud.rs]: Renamed it to `order=ascending`/`order=descending`.

File: server/svix-server/src/v1/utils/mod.rs (hunk 4 above)
Comment (svix-andor) [reply server/svix-server/src/v1/utils/mod.rs]: I made `apply_pagination` and `list_response` generic again, they simply take the ordering as a parameter.

File: server/svix-server/src/v1/endpoints/endpoint/crud.rs (hunk 2 above)
Comment (svix-andor) [reply server/svix-server/src/v1/endpoints/endpoint/crud.rs]: Resolving this since it's not a boolean anymore, therefore not relevant.

**Review by tasn — APPROVED** [review]
I think it's fine. Just making sure: you have the tests to ensure this works + tests to verify you haven't broken the ordering for anything else?

**Comment by svix-andor** [comment]
I believe so. The again-generic `apply_pagination` and `listing_response` is only used for Endpoint listing, the remainders still use the direction specific variant (which I intend to remove once all have bidirectional listing).
